krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -106,60 +106,75 @@ func (r *itfce) GetGoStruct() (*nephioreqv1alpha1.Interface, error) {
 // GetAttachmentType returns the attachmentType from the spec
 // if an error occurs or the attribute is not present an empty string is returned
 func (r *itfce) GetAttachmentType() string {
+	if r.o == nil {
+		return ""
+	}
 	return r.getStringValue(attachmentType...)
 }
 
 // GetCNIType returns the cniType from the spec
 // if an error occurs or the attribute is not present an empty string is returned
 func (r *itfce) GetCNIType() string {
+	if r.o == nil {
+		return ""
+	}
 	return r.getStringValue(cniType...)
 }
 
 // GetNetworkInstanceName returns the name of the networkInstance from the spec
 // if an error occurs or the attribute is not present an empty string is returned
 func (r *itfce) GetNetworkInstanceName() string {
+	if r.o == nil {
+		return ""
+	}
 	return r.getStringValue(networkInstanceName...)
 }
 
 // SetAttachmentType sets the attachmentType in the spec
 func (r *itfce) SetAttachmentType(s string) error {
-
-	// validation -> should be part of the interface api repo
-	switch s {
-	case string(nephioreqv1alpha1.AttachmentTypeNone):
-	case string(nephioreqv1alpha1.AttachmentTypeVLAN):
-	default:
+	if r.o == nil {
+		return fmt.Errorf(errKubeObjectNotInitialized)
+	}
+	if !nephioreqv1alpha1.IsAttachmentTypeSupported(s) {
 		return fmt.Errorf("unknown attachmentType")
 	}
 
-	return r.setNestedField(s, attachmentType...)
+	return r.o.SetNestedField(s, attachmentType...)
 }
 
 // SetCNIType sets the cniType in the spec
 func (r *itfce) SetCNIType(s string) error {
```

Comment:
Same as other setter. Call the common function.

Reply:
aligned